Packs/MicrosoftExchangeOnline/Integrations/EWSO365/EWSO365_test.py

```diff
@@ -823,4 +837,30 @@ def test_create_message_with_html_body(self, mock_message, mock_body, mock_html_
             name="file.txt", content="data", is_inline=True, content_id="12345"
         )
         mock_message.assert_called_once()
-        self.assertIsInstance(result, MagicMock)
+        assert isinstance(result, MagicMock)
+
+
+@pytest.mark.parametrize("headers, expected_formatted_headers", [
+    ([("Message-ID", '<valid_header>')], [("Message-ID", '<valid_header>')]),
+    ([("Message-ID", '<[valid_header]>')], [("Message-ID", '<valid_header>')]),
+    ([("Message-ID", 'Other type of header format')], [("Message-ID", 'Other type of header format')]),
+])
+def test_handle_attached_email_with_incorrect_id(mocker, headers, expected_formatted_headers):
+    """
+    Given:
+        - case 1: valid Message-ID header value in attached email object
+        - case 1: invalid Message-ID header value in attached email object
+        - case 3: a Message-ID header value format which is not tested in the context of handle_attached_email_with_incorrect_id
+    When:
+        - fetching email which have an attached email with Message-ID header
+    Then:
+        - case 1: verify the header in the correct format and return the attached email object
+        - case 2: check if the Message-ID header value is in the correct format, if not, fix it if possible.
+        - case 3: return the header value without without further handling
```

Comment:
Only a suggestion - not required for merging:

- Use pytest.params(..., id=valid header / invalid_header / untouched format)
- simplify the test description

Or just leave it as is 🙂

Reply:
Wasn't familiar with `pytest.params`'s `id` argument. love it!
I've simplified the description a a bit, let me know if you'd do something different.